src/websocket_tracker_connection.cpp

```diff
@@ -254,14 +254,16 @@ void websocket_tracker_connection::do_send(tracker_answer const& ans)
 	obj["sdp"] = ans.answer.sdp;
 
 	json::string const data = json::to_string(payload);
+	m_write_data.assign(data.begin(), data.end());
```

Comment:
I take it there's never more than one outstanding send, right?

Reply:
No, there is a send queue and messages are sent one by one.

Reply:
if you do:
```
m_write_data = json::to_string(payload);
```

The heap allocation will get moved into the member function, instead of being copied.

Also, do you really need to namespace-qualify the call to `to_string()`? I would think it would be picked up by ADL. But maybe it's nice to have it be explicit

Reply:
Sadly a `boost::json::string` is not an `std::string`, so this would require making `m_write_data` a `json::string` and moving the `boost.json` inclusion in the header.

Reply:
I see